[Zeepheru] iP

Zeepheru opened the pull request:
# This is a heading

- This is a bullet list
- This is a bullet list

1. This is a numbered list
2. This is a numbered list

```java
// This is a fenced code block
System.out.println("Hello world!");
```

- [ ] This is a task list
- [ ] This is a task list

These are emojis: 😅😭🔥🔥

> This is 
> a
> block quote

[This is a hyperlink](https://www.youtube.com/watch?v=dQw4w9WgXcQ)

This is some inline code `foo();`

Here is formatted text: **bold** | *italic* | ~~strikethrough~~

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Zeepheru committed:
.gitignore: Update

More extensive gitignore that, importantly, covers files generated
by Java (e.g. .class).

Zeepheru committed:
Replace all instances of D*ke with the new name Torne.

Torne is the superior name.
D*ke is never to be used as is a bannable offence.

Zeepheru committed:
Add greeting and exit

Zeepheru committed:
Add echo and exit funtionality

The bot now runs in a loop, echoing all inputs
and exiting on a ye input.

Zeepheru committed:
Add basic task handling code

Zeepheru committed:
TaskHandler.java: Add getting number of tasks

Zeepheru committed:
ChatOutput.java: Add error text output

Zeepheru committed:
Add addTask and list functionality, add command parsing

Command parsing is written to easily integrate and accept future
styles of arguments.

Zeepheru committed:
Add task completion tracking

Zeepheru committed:
Torne.java: edit comments

Zeepheru committed:
Add subclasses of Task

Subclasses that correspond to different types of Task

Zeepheru committed:
Add Torne support for adding different types of tasks

Zeepheru committed:
Torne.java: Correct arg flag from \ to /

As well as other output string formatting improvements.

Zeepheru committed:
TaskEvent.java: Correct task code string from D to E

Zeepheru committed:
Torne.java: Add JavaDocs comments

Zeepheru committed:
Torne.java: Correct JavaDocs comment formatting

Zeepheru committed:
Add help command that displays all commands

Zeepheru committed:
Add ability to delete tasks by index

Zeepheru committed:
Add error handling for invalid commands

Zeepheru committed:
Update border line

Zeepheru committed:
Update border line

Zeepheru committed:
Add io redirection testing

Zeepheru committed:
Torne.java: Rename taskHandler since it's no longer a constant

Zeepheru committed:
Add saving of task data to a local file

Tasks now have a separate String output formatted for local storage.

Zeepheru committed:
Add ability to load and save tasks to local storage

WinstonJin commented on src/main/java/ChatOutput.java:
```diff
@@ -0,0 +1,33 @@
+public class ChatOutput {
+    private static final String BORDER_LINE = "-------------------------------------------------------------";
+    public static final String INDENT = "    ";
+
```
Awesome work on the SCREAMING_SNAKE_CASE format

WinstonJin commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,101 @@
+            throw new TorneInvalidDataException("Input task string does not have enough components");
+        }
+
+        boolean isTaskDone;
```
Nice naming of the boolean isDone as well as the Exceptions. The naming of the InvalidData and InvalidCommand Exceptions are clear to understand

WinstonJin commented on src/main/java/Torne.java:
```diff
@@ -0,0 +1,398 @@
+    private static Storage storage;
+    private static final String[] NO_ARGS = new String[0];
+    private static final String[] DEFAULT_ARG = {""};
+    private static final Map<String, String[]> COMMANDS = Map.of(
```
Awesome job in creating a Map that effectively reduces the number of classes you have to make (Command class etc)

WinstonJin commented on src/main/java/Torne.java:
```diff
@@ -0,0 +1,398 @@
+            return;
+        }
+
+        StringBuilder helpMessageBuilder = new StringBuilder("This is the list of Torne commands and options:");
```
Really nice help message!

WinstonJin reviewed (commented):
Overall, really good code quality naming!

LeongWZ commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,101 @@
+
+        // parse second component "1" / "0" for completion status
+        switch (parts[1]) {
+        case "1":
```
Switch case indentation follows Coding Standard. Good!

LeongWZ commented on src/main/java/Torne.java:
```diff
@@ -0,0 +1,398 @@
+import java.util.*;
```
Avoid wildcard import (*) as indicated by the Coding Standard. Do rectify this by specifying the imported classes.

LeongWZ commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,101 @@
+            throw new TorneInvalidDataException("Storage string task type code invalid.");
+        }
+
+        // set completion status
```
Good use of comments in this method. This makes the code easier to follow. Good!

LeongWZ commented on src/main/java/Torne.java:
```diff
@@ -0,0 +1,398 @@
+     */
+    private void showExitMessage() {
+        String exitText = """
+Aww, bye to you as well :c""";
```
Line 161 is not indented correctly. Use double tabs to indent this line.

LeongWZ commented on src/main/java/Torne.java:
```diff
@@ -0,0 +1,398 @@
+ */
+
+public class Torne {
+    private static final ChatOutput OUTPUT = new ChatOutput();
```
Static constants are in all caps as indicated by the Coding Standard. Good!

LeongWZ reviewed (commented):
Good use of comments which made code easy to read and follow along. Coding standard is adhered to most of the time, though there are a few nits which need to be rectified. Good job!

Zeepheru committed:
Add date-time support

Improve task loading sequence such that loading from commands
and from storage is differentiated.
Add TorneDateTime class to handle all dateTime parsing and
formatting operations specific to the program.

Zeepheru committed:
Organize classes into packages

Zeepheru committed:
Task.Java: Change Task to protected

Zeepheru committed:
Rewrite command structure

- Commands are now their own class with more easily callable properties.
- Parsing is done by a separate class.
- Command arguments are represented by their own classes as well for ease of display.

Zeepheru committed:
Merge branch 'add-gradle-support'

Zeepheru committed:
DeleteTask.java: Correct command message

Zeepheru committed:
TorneDateTime.java: Fix relative string computation

Zeepheru committed:
Add iniital unit tests

Zeepheru committed:
Add find feature

Zeepheru committed:
Torne.java: Clean up file

Delete redundant fields.
Set fields for Torne features that are effectively inal to inal.

Zeepheru committed:
Torne.java: Delete more redundant fields

Zeepheru committed:
README.md: Update

Zeepheru committed:
ChatOutput.java: Correct indentation

Zeepheru committed:
Merge branch 'master' of https://github.com/Zeepheru/ip

Zeepheru committed:
Improve code styling

Various files have minor issues with regards to code styling:
- Commented unused imports
- Missing or extra empty lines between code elements
- Unindented hanging text as part of a string

Most of them do not specifically violate code quality guidelines, but do irritate my OCD. Especially minor pain points like leaving an empty line before the end of a block, grr. Removing these minor issues satiates my OCD.

It's preferable than leaving it in its current state because my OCD says so.

Zeepheru committed:
Implement basic JavaFX UI

Copy code from the JavaFX tutorial to implement a basic, but working version of a Chat Bot UI. No configuration is changed except that necessary to have it work with Torne.

Modify Torne such that it still can run as a CLI, but ChatOutput also stores the last Torne output so a GUI can fetch it and display it as a message. This also means that, as the GUI is running, Torne also prints the equivalent commands to the terminal.

Zeepheru committed:
Merge pull request #3 from Zeepheru/branch-Level-10

Implement basic JavaFX UI

Zeepheru committed:
Configure Gradle build settings

Building using the CLI doesn't work properly, had to fix this by forcing Gradle version 8.2
in gradle-wrapper.properties.

Change the name of the main class in build.gradle to the one used by the GUI.

Also set up everything so it builds a fat Jar file with JavaFX.

Zeepheru committed:
Change GUI colors

Original looks pretty meh.

Change to something with at least a similar theme
(orange-red, for some reason).

Zeepheru committed:
Make improvements to UI

Zeepheru committed:
Remove border lines from GUI output

Zeepheru committed:
Remove border lines from GUI output

Zeepheru committed:
Fix border lines

Zeepheru committed:
improve UI

To further improve the Torne UI
and customize it.

Scaling of the scrollpane is broken,
fix by anchoring pane to bottom and
right of window.

Replace Duke image with Torne image
that I drew.

Also various comments and method names
refer to Duke, rename to Torne.

Zeepheru committed:
Merge branch 'master' into branch-Level-10

Zeepheru committed:
Merge pull request #1 from Zeepheru/branch-A-CodeQuality

Improve code styling

Zeepheru committed:
Merge pull request #5 from Zeepheru/branch-Level-10

GUI Improvements

Zeepheru committed:
MainWindow.fxml: reformat and change JavaFX version

Zeepheru committed:
Modify Torne response loop

Save to storage after every response instead of at the end

Zeepheru committed:
Fix shutdown procedure

Torne doesn't close itself on a 'bye'
command as it should.

Rectify this by having a check in the
UI for the input 'bye', which calls
a Torne shutdown and closes the
application.

Zeepheru committed:
Add support for error-formatted responses

Responses are passed to the UI as strings. This
makes it difficult for errors to be passed
and specially formatted.

Create new ChatResponse class to encapsulate
the response string as well as response-related
information - in this case whether it is an
error message. So when it is passed to the UI
the UI can determine what kind of formatting
needs to be applied to the response message.

Zeepheru committed:
Add custom formatting for Torne messsages

Zeepheru committed:
Add app icon

Source: https://www.youtube.com/watch?v=UZKKaI8OnjY

Zeepheru committed:
Modify intro text

Zeepheru committed:
Add Ui screenshot

Zeepheru committed:
Add additional index checking for task commands

Index checking is handled within each command class
-> duplicated code.

Refactor index checking out into TaskHandler.

Add additional error messages for specific index
errors.

Zeepheru committed:
README.md: update

Zeepheru committed:
Change default window size to better fit 1920x1080

Zeepheru committed:
README.md: correct version badge

Zeepheru committed:
Update docs readme

Zeepheru committed:
Add smaller Logo to be used for README

Zeepheru committed:
Add image

Zeepheru committed:
Remove dead code

Zeepheru committed:
Remove emojis from docs README

They don't work.

Zeepheru committed:
Change find command to case-insensitive

Zeepheru committed:
Correct type in docs readme

Zeepheru committed:
Update README.md

Zeepheru committed:
Update readme

Zeepheru committed:
Update readme

Zeepheru committed:
Add assertions

Zeepheru committed:
Merge pull request #6 from Zeepheru/branch-A-Assertions

Assertions

Zeepheru committed:
Reformat all code

Remove unnecessary imports,
change JavaDocs params.

Reformat operation done using Intellij's
buit-in reformatter

Zeepheru committed:
Merge branch 'master' of https://github.com/Zeepheru/ip

Zeepheru committed:
Add test cases
